### ISSUE TITLE
Token authentication not working with prosody 0.11.0

### ISSUE THREAD

**Issue by sapkra** [jitsi/lib-jitsi-meet#837]
## Description
I tried using the new prosody version 0.11.0 but I'm getting the following errer when I'm trying to login with my token. Everything ist working fine with prosody-trunk version 747.

Would be great if we can find out the reason and fix it soon as possible. I think the bosh problem which was fixed in trunk is now also fixed in the new release.

## Current behavior
Getting this error:
```
general      warn	Error verifying token err:not-allowed, reason:token required
```
No prosody.log or prosody.err file is created. The directory `/var/log/prosody` is empty.

## Expected Behavior
The login with the token should work like expected & of cause without any errors.

## Steps to reproduce
Just update prosody to 0.11.0.

## Possible solution
I find out that the path for the patch for `mod_bosh` is not exactly the same like in the prosody release. Maybe that's the problem?

Patch from jitsi-meet repository
```
(hosts[session.host]).events.fire_event(eventName, { session = session, request = request });
```

Code from prosody 0.11.0
```
module:fire_event("bosh-session", { session = session, request = request });
```
# Environment details
I'm running everything in docker containers based on the official `docker-jitsi-meet` images.

**Comment by sconvent** [comment]
I had the same issue and changing the line
```
module:fire_event("bosh-session", { session = session, request = request });
```
to 
```
hosts[session.host].events.fire_event("bosh-session", { session = session, request = request });
```
solved the issue for me, as suggested under Possible Solution.
However, I don't fully understand why.

**Comment by saghul** [comment]
I think we need to change this: https://github.com/jitsi/jitsi-meet/blob/54c36198d060bfaa8e88d4594066a44310a579c3/resources/prosody-plugins/mod_auth_token.lua#L21 to say `hook_global` since `mod_bosh` is a global module: https://hg.prosody.im/0.11/file/tip/plugins/mod_bosh.lua#l9

@sapkra if you have the setup ready, can you give that change a try?

**Comment by sapkra** [comment]
Yeah, I tested it and it works. 🎉

**Comment by damencho** [comment]
@sapkra nice, thank you for testing.

@saghul I'm wondering how to make the module work in both cases 0.11 and versions before it?

**Comment by saghul** [comment]
@damencho I think we can just switch to `hook_global` and modify our patch. WDYT? Also, since the patch is not available on Prosody 0.11, we can point people at the official repo to install it, instead of patching trunk.

I'll make a PR with this change and we can discuss it there.

**Comment by saghul** [comment]
> Yeah, I tested it and it works. 🎉

Thank you so much for testing!
